File: Army/units/vampire.cpp
#include "vampire.h"

using namespace std;

Vampire::Vampire(const string& name, int healthPoint, int damage) : Unit(name, healthPoint, damage)  {
    setAbility(new VampireAbility(this));
    setCurrentState(new HumanState(this));
    setNextState(new BatState(this));
    setUnitType(vampire);
    setIsDead();
}

Vampire::~Vampire() {}

File: Army-Java/Units/Vampire.java
package com.gymfox.Army.Units;

import com.gymfox.Army.Ability.VampireAbility;

public class Vampire extends Unit {
    public Vampire(String name, int healthPointLimit, int damage) {
        super(name, healthPointLimit, damage);
        this.ability = new VampireAbility(this);
        this.isDead = true;
    }
}
